spec/controllers/user_reports_controller_spec.rb

```diff
@@ -0,0 +1,158 @@
+require "rails_helper"
+
+RSpec.describe UserReportsController, type: :controller do
+  let(:user){FactoryBot.create :user}
+  let(:user2){FactoryBot.create :user, full_name: "Name 2", email: "test2@mail.com"}
+  let(:user3){FactoryBot.create :user, full_name: "Name 3", email: "test3@mail.com"}
+  let(:course){FactoryBot.create :course}
+  let!(:user_course){FactoryBot.create :user_course, course: course, user: user}
+  let!(:user_course2){FactoryBot.create :user_course,
+                        course: course, user: user2}
+  let(:report_1){FactoryBot.create :user_report, content: "The first report",
+                   date: 2.days.ago, course: course, user: user}
+  let(:report_2){FactoryBot.create :user_report, content: "The first report",
+                   date: 2.days.ago, course: course, user: user2}
+
+  before{login user}
+
+  describe "GET /index" do
+    context "when user has courses" do
+      before{get :index}
+
+      it {expect(response).to render_template :index}
+    end
+
+    context "when user don't have course" do
+      before do
+        login user3
+        get :index
+      end
+
+      it {expect(response).to redirect_to root_url}
+    end
+  end
+
+  describe "GET /new" do
+    before{get :new}
+
+    it {expect(response).to render_template :new}
```

Comment:
sau it nên ghi đoạn mô tả mong muốn e trả về là gì nhé
vd:
it "render template new" do 
...
end

Reply:
tương tự các cái khác

Reply:
Vâng ạ.